ibis/mapd/client.py

```diff
@@ -923,10 +929,15 @@ def sql(self, query):
         -------
         table : TableExpr
         """
-        # Get the schema by adding a LIMIT 0 on to the end of the query. If
-        # there is already a limit in the query, we find and remove it
-        limited_query = 'SELECT * FROM ({}) t0 LIMIT 1'.format(query)
-        schema = self._get_schema_using_query(limited_query)
+        if pymapd_dtype is None:
+            raise com.UnsupportedOperationError(
+                'This method is available just on Python version >= 3.6.'
+            )
+        # Remove `;` + `--` (comment)
+        query = re.sub(r'\s*;\s*--', '\n--', query.strip())
+        # Remove trailing ;
+        query = re.sub(r'\s*;\s*$', '', query.strip())
```

Comment:
Why do we need to do this regular expression munging? There's a constraint on the query passed to the `sql` method, which is that it must produce a relation. Why does it matter whether there are comments or anything else in query as long as the query is syntactically valid and it ends in a relation-producing expression?

Reply:
As regex substitution could be `dangerous` I am trying to keep it pretty simple to avoid replace undesired text.

the most important is `query = re.sub(r'\s*;\s*$', '', query.strip())` it is pretty simple and objective ... as I can see, there is no side effect on this.

but in some cases after `;` the uses maybe add comments `--` and that is why I need to add this extra pattern to treat specifically when there is `;` followed by `--`.

I think these 2 regex tries to keep changing as minimum as possible.